core/test/gather/gatherers/full-page-screenshot-test.js

```diff
@@ -202,4 +208,338 @@ describe('FullPageScreenshot gatherer', () => {
+     * @param {DebugFormat} debugFormat
+     * @return {Promise<NodeAnalysisResult>}
+     */
+    function analyzeScreenshotNodes(page, lhr, debugFormat) {
+      const options = {
+        fullPageScreenshot: lhr.fullPageScreenshot,
+        // We currently can't get lossless encodings here, so we must allow for some difference.
+        // I found 2.4 is what I get with the test data for regions that match, and 100+ for regions that are clearly not matching.
+        // So using 5 seems a good bet.
+        // If we could get lossless webp/png here, this should go away.
+        // TODO https://bugs.chromium.org/p/chromium/issues/detail?id=1469183
+        individualPixelColorThreshold: 5,
+        debugFormat,
+      };
+
+      return page.evaluate(async (options) => {
+        function hexToRgb(hex) {
+          const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+          return {
+            r: parseInt(result[1], 16),
+            g: parseInt(result[2], 16),
+            b: parseInt(result[3], 16),
+          };
+        }
+
+        // https://www.compuphase.com/cmetric.htm
+        function colorDistance(e1, e2) {
+          const rmean = (e1.r + e2.r) / 2;
+          const r = e1.r - e2.r;
+          const g = e1.g - e2.g;
+          const b = e1.b - e2.b;
+          return Math.sqrt(
+            (((512 + rmean) * r * r) >> 8) + 4 * g * g + (((767 - rmean) * b * b) >> 8));
+        }
+
+        const img = await new Promise((resolve, reject) => {
+          // eslint-disable-next-line no-undef
+          const img = new Image();
+          img.onload = () => resolve(img);
+          img.onerror = reject;
+          img.src = options.fullPageScreenshot.screenshot.data;
+        });
+
+        // eslint-disable-next-line no-undef
+        const canvasEl = document.createElement('canvas');
+        canvasEl.width = img.width;
+        canvasEl.height = img.height;
+        const ctx = canvasEl.getContext('2d');
+        ctx.drawImage(img, 0, 0);
+
+        const results = [];
+        for (const node of Object.values(options.fullPageScreenshot.nodes)) {
+          if (!(node.id.includes('green') || node.id.includes('red'))) {
+            continue;
+          }
+          if (!node.id.includes('el-')) {
+            continue;
+          }
+
+          const expectedColor = hexToRgb(node.id.includes('green') ? '#427f36' : '#8a4343');
+          const result = {id: node.id, success: true};
+          results.push(result);
+
+          const right = Math.min(node.right, canvasEl.width - 1);
+          const bottom = Math.min(node.bottom, canvasEl.height - 1);
+          const width = right - node.left;
+          const height = bottom - node.top;
+
+          let pixelDifferCount = 0;
+          const debugData = [];
+          for (let y = node.top; y <= bottom; y++) {
+            const row = [];
+            debugData.push(row);
+
+            for (let x = node.left; x <= right; x++) {
```

Comment:
This seems a lot like a re-implementation of existing pixel matchers. Could we leverage something like [`expect-mocha-image-snapshot`](https://www.npmjs.com/package/expect-mocha-image-snapshot) to do the image diffs for us? We use something similar in LHCI and I found it very helpful to review image diffs in Github that way.